clients/pkg/promtail/targets/file/filetargetmanager.go

```diff
@@ -307,9 +307,14 @@ func (s *targetSyncer) sync(groups []*targetgroup.Group, targetEventHandler chan
 			}
 
 			level.Info(s.log).Log("msg", "Adding target", "key", key)
-			watcher := make(chan fsnotify.Event)
-			s.fileEventWatchers[string(path)] = watcher
-			t, err := s.newTarget(string(path), labels, discoveredLabels, watcher, targetEventHandler)
+
+			wkey := string(path)
+			watcher, ok := s.fileEventWatchers[wkey]
```

Comment:
In which case is there already a watcher?

Reply:
There should not be a case, but it makes sense to check anyway.